share/dkg/dkg_test.go

```diff
@@ -0,0 +1,616 @@
+	cc, err = dkg2.ProcessSecretCommits(sc)
+	assert.Nil(t, cc)
+	assert.Error(t, err)
+	sc.SessionID = goodSid
+
+	// wrong commitments
+	goodPoint := sc.Commitments[0]
+	sc.Commitments[0] = suite.Point().Null()
+	msg = sc.Hash(suite)
+	sig, err := sign.Schnorr(suite, dkg.long, msg)
+	require.Nil(t, err)
+	goodSig = sc.Signature
+	sc.Signature = sig
+	cc, err = dkg2.ProcessSecretCommits(sc)
+	assert.NotNil(t, cc)
+	assert.Nil(t, err)
+	sc.Commitments[0] = goodPoint
+	sc.Signature = goodSig
+
+	// all fine
+	cc, err = dkg2.ProcessSecretCommits(sc)
+	assert.Nil(t, cc)
+	assert.Nil(t, err)
+}
+
+func TestDKGComplaintCommits(t *testing.T) {
+	fullExchange(t)
+
+	var scs []*SecretCommits
+	for _, dkg := range dkgs {
+		sc, err := dkg.SecretCommits()
+		require.Nil(t, err)
+		scs = append(scs, sc)
+	}
+
+	for _, sc := range scs {
+		for _, dkg := range dkgs {
+			cc, err := dkg.ProcessSecretCommits(sc)
+			assert.Nil(t, err)
+			assert.Nil(t, cc)
+		}
+	}
+
+	// change the sc for the second one
+	wrongSc := &SecretCommits{}
+	wrongSc.Index = scs[0].Index
+	wrongSc.SessionID = scs[0].SessionID
+	wrongSc.Commitments = make([]abstract.Point, len(scs[0].Commitments))
+	copy(wrongSc.Commitments, scs[0].Commitments)
+	//goodScCommit := scs[0].Commitments[0]
+	wrongSc.Commitments[0] = suite.Point().Null()
+	msg := wrongSc.Hash(suite)
+	wrongSc.Signature, _ = sign.Schnorr(suite, dkgs[0].long, msg)
+
+	dkg := dkgs[1]
+	cc, err := dkg.ProcessSecretCommits(wrongSc)
+	assert.Nil(t, err)
+	assert.NotNil(t, cc)
+
+	dkg2 := dkgs[2]
+	// ComplaintCommits: wrong index
+	goodIndex := cc.Index
+	cc.Index = uint32(nbParticipants)
+	rc, err := dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+	cc.Index = goodIndex
+
+	// invalid signature
+	goodSig := cc.Signature
+	cc.Signature = randomBytes(len(cc.Signature))
+	rc, err = dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+	cc.Signature = goodSig
+
+	// no verifiers
+	v := dkg2.verifiers[uint32(0)]
+	delete(dkg2.verifiers, uint32(0))
+	rc, err = dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+	dkg2.verifiers[uint32(0)] = v
+
+	// deal does not verify
+	goodDeal := cc.Deal
+	cc.Deal = &vss.Deal{
+		SessionID:   goodDeal.SessionID,
+		SecShare:    goodDeal.SecShare,
+		RndShare:    goodDeal.RndShare,
+		T:           goodDeal.T,
+		Commitments: goodDeal.Commitments,
+		Signature:   randomBytes(len(goodDeal.Signature)),
+	}
+	rc, err = dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+	cc.Deal = goodDeal
+
+	//  no commitments
+	sc := dkg2.commitments[uint32(0)]
+	delete(dkg2.commitments, uint32(0))
+	rc, err = dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+	dkg2.commitments[uint32(0)] = sc
+
+	// secret commits are passing the check
+	rc, err = dkg2.ProcessComplaintCommits(cc)
+	assert.Nil(t, rc)
+	assert.Error(t, err)
+
+	// TODO find a way to be the malicious guys,i.e.
```

Comment:
I actually don't know how to make this error happening so I let it as is for the moment.